### PR TITLE
feat: change user property name to createdBy

### PR DESCRIPTION
**Pull request by vietnguyen**
- Removed the HibernateListener solution as it may cause performance issue.
- Added code to ignore `createdBy` field in `IdentifiableObjectBundleHook.preUpdate`
- Test covered by `MetadataImportServiceTest#testUpdateImmutableCreatedByField`

Screen Shot 2021-01-13 at 7 23 50 PM

### CODE REVIEW

**Review by mortenoh — APPROVED** [review]

**Review by stian-sandvold — APPROVED** [review]

**Comment by stian-sandvold** [comment]
Good work @vietnguyen . I see a few automatic comments from github in the files, did you have a look at them? Some seem sensible to check out.

There is also 1 "critical" issue from SonarCloud check, related to using a constant for "createdBy" in one of the files, would be great to add that to please sonarCloud :)

**Comment by larshelge** [comment]
Let's make the `createdBy` property unmodifiable / immutable.

**Review by larshelge — CHANGES_REQUESTED** [review]
Let's make the `createdBy` field immutable / not modifiable.

**Comment by larshelge** [comment]
Remember to upgrade the change note and share with front-end team.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/adapter/BaseIdentifiableObject_.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.common.adapter;/*
+                                     * Copyright (c) 2004-2021, University of Oslo
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/adapter/BaseIdentifiableObject_.java:29]: Remove license.

Comment (vietnguyen) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/adapter/BaseIdentifiableObject_.java]: fixed

**Comment by vietnguyen** [comment]
> Let's make the `createdBy` field immutable / not modifiable.
@larshelge 
- Removed the HibernateListener solution as it may cause performance issue.
- Added code to ignore `createdBy` field in `IdentifiableObjectBundleHook.preUpdate`
- Test covered by `MetadataImportServiceTest#testUpdateImmutableCreatedByField`

**Pull request merged**